src/util/git/repoRef.ts

```diff
@@ -37,25 +36,42 @@ export function repoRefFromPush(push: OnPushToAnyBranch.Push) {
                 branch: push.branch,
             });
         case ProviderType.bitbucket :
-            const url = push.repo.org.provider.url.replace("http://", "");
-            const id = new BitBucketServerRepoRef(
-                url,
-                push.repo.owner,
-                push.repo.name,
-                true,
-                push.after.sha,
-            );
-            id.branch = push.branch;
-            id.cloneUrl = (creds: BasicAuthCredentials) =>
-             `http://${encodeURIComponent(creds.username)}:${encodeURIComponent(creds.password)}@${id.remoteBase}${id.pathComponent}.git`;
-            return id;
+            const providerUrl = push.repo.org.provider.url;
+            return toBitBucketServerRepoRef({
+                providerUrl,
+                owner: push.repo.owner,
+                name: push.repo.name,
+                sha: push.after.sha,
+                branch: push.branch,
+            });
         case ProviderType.bitbucket_cloud :
             throw new Error("BitBucket Cloud not yet supported");
         default:
             throw new Error(`Provider ${providerType} not currently supported in SDM`);
     }
 }
 
+export function toBitBucketServerRepoRef(params: {
+    providerUrl: string,
+    owner: string,
+    name: string,
+    sha: string,
+    branch?: string,
+}): BitBucketServerRepoRef {
+    const url = params.providerUrl;
+    const id = new BitBucketServerRepoRef(
+        url,
+        params.owner,
+        params.name,
+        true,
```

Comment:
this (I'm guessing it's isProject) is going to need to come from the graph, we should make an issue at least

Reply:
Ben says we don't currently support users so I guess it doesn't matter